src/rules/no-extraneous-dependencies.js

```diff
@@ -1,16 +1,22 @@
 import path from 'path'
+import fs from 'fs'
 import readPkgUp from 'read-pkg-up'
 import minimatch from 'minimatch'
 import importType from '../core/importType'
 import isStaticRequire from '../core/staticRequire'
 
-function getDependencies(context) {
+function getDependencies(context, packagePath) {
   try {
-    const pkg = readPkgUp.sync({cwd: context.getFilename(), normalize: false})
-    if (!pkg || !pkg.pkg) {
+    const pkg = packagePath
+      ? JSON.parse(fs.readFileSync(packagePath, 'utf8'))
```

Comment:
You could report the error at the beginning of the file, like [here](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/no-abusive-eslint-disable.js#L14-L20).

I would advise against throwing an error, as ESLint doesn't properly handle them, the process will crash and the error will be hard to understand. By reporting it this way, the user will get an error for pretty much every file, but it will be much clearer and visible than a wondering `console.error`.

```js
 if (packagePath && err.code === 'ENOENT') {
    // find --> found
      context.report({ message: 'The package.json file could not be found.' })
    }
    if (packagePath && err instanceof SyntaxError) {
      // Change the error message to this and append the error message
      context.report({ message: 'The package.json file could not be parsed: ' + err.message})
    }
```